multicluster/controllers/multicluster/leader/member_cleanup_controller.go

```diff
@@ -0,0 +1,184 @@
+				return ctrl.Result{}, err
+			}
+			cleanupSucceed := c.cleanupResources(ctx, clusterID, allResExports)
+			if !cleanupSucceed {
+				return ctrl.Result{}, fmt.Errorf("failed to clean up all stale ResourceExports for the member cluster %s, retry later", clusterID)
+			}
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, nil
+}
+
+func (c *MemberResourcesCleanupController) getAllResourceExports(ctx context.Context) ([]mcv1alpha1.ResourceExport, error) {
+	resourceExports := &mcv1alpha1.ResourceExportList{}
+	err := c.Client.List(ctx, resourceExports, &client.ListOptions{Namespace: c.namespace})
+	if err != nil {
+		return nil, err
+	}
+	return resourceExports.Items, nil
+}
+
+func (c *MemberResourcesCleanupController) checkAndCleanup(ctx context.Context) {
+	validMemberClusterIDs, err := c.getValidMemberClusterIDs(ctx)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get valid member cluster's ClusterID in the Namespace", "namespace", c.namespace)
+		return
+	}
+
+	allResExports, err := c.getAllResourceExports(ctx)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get ResourceExports in the Namespace", "namespace", c.namespace)
+		return
+	}
+
+	allMemberClusterIDs, err := c.getAllMemberClusterIDs(ctx, allResExports)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get all member cluster's ClusterID in the Namespace", "namespace", c.namespace)
+		return
+	}
+
+	staleMemberClusterIDs := allMemberClusterIDs.Difference(validMemberClusterIDs)
+	for clusterID := range staleMemberClusterIDs {
+		cleanupSucceed := c.cleanupResources(ctx, clusterID, allResExports)
+		if cleanupSucceed {
+			klog.InfoS("The member cluster's stale resources are cleaned up", "clusterID", clusterID)
+		} else {
+			klog.Error("Failed to clean up the member cluster's stale resources", "clusterID", clusterID)
+		}
+	}
+}
+
+func (c *MemberResourcesCleanupController) getAllMemberClusterIDs(ctx context.Context, resourceExports []mcv1alpha1.ResourceExport) (sets.Set[string], error) {
+	allMemberClusterIDs := sets.Set[string]{}
+	for _, resourceExport := range resourceExports {
+		if resourceExport.Spec.ClusterID != "" {
+			allMemberClusterIDs.Insert(resourceExport.Spec.ClusterID)
+		}
+	}
+	return allMemberClusterIDs, nil
+}
+
+func (c *MemberResourcesCleanupController) getValidMemberClusterIDs(ctx context.Context) (sets.Set[string], error) {
+	validMemberClusterIDs := sets.Set[string]{}
+	memberClusterAnnounces := &mcv1alpha1.MemberClusterAnnounceList{}
+	err := c.Client.List(ctx, memberClusterAnnounces, &client.ListOptions{Namespace: c.namespace})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, m := range memberClusterAnnounces.Items {
+		validMemberClusterIDs.Insert(m.ClusterID)
+	}
+	return validMemberClusterIDs, nil
+}
+
+func (c *MemberResourcesCleanupController) cleanupResources(ctx context.Context, clusterID string, resouceExports []mcv1alpha1.ResourceExport) bool {
```

Comment:
`cleanUpResources` or `cleanUpResourceExports `(if we just remove Exports, not other resources).